### ISSUE TITLE
[BUG] Filter return type

### ISSUE THREAD

**Issue by hlynurf** [vemel/mypy_boto3_builder#8]
**Describe the bug**
Return type from the filter function is not annotated but in 1.12.2.0 it was annotated as mypy_boto3_s3.service_resource.BucketObjectsCollection.

**To Reproduce**
1. Install 'boto3-stubs[...]'
2.  Run mypy on the following code sample

```python
import boto3


resource = boto3.session.Session(region_name='eu-central-1').resource(
	service_name='s3',
	aws_access_key_id='xxx',
	aws_secret_access_key='xxx'
)
bucket = resource.Bucket('bucket-name')
reveal_type(bucket.objects.filter(Prefix='prefix'))

```
3. Actual mypy output

```
note: Revealed type is '<nothing>'
```

**Expected mypy-output**

```
note: Revealed type is 'mypy_boto3_s3.service_resource.BucketObjectsCollection'
```

**Additional context**
OS: Ubuntu 18.04
boto3 installation method: pip
boto3 version: 1.12.16

**Comment by vemel** [comment]
Looks like `TypeVar` approach is not working for collections. I have a fix already, just have to rollback collection return type changes. Thank you for the report!

**Comment by vemel** [comment]
Should be fixed in [1.0.5](https://github.com/vemel/mypy_boto3_builder/releases/tag/1.0.5-release). I updated E2E tests to include your sample. Please update to `boto3-stubs>=1.12.17.1` and update `boto3>=1.12.17`.

Please let me know if the new version is working as expected. Thank you for your help!

**Comment by hlynurf** [comment]
Thanks for the quick response. This fixed it.
